features/facebookImport/src/views/ministories/reportStory.jsx

```diff
@@ -0,0 +1,40 @@
+import Story from "./story.jsx";
+
+class ReportStory extends Story {
+    constructor(account) {
+        super(account);
+        this._reports = account.reports;
+        this._neededReports = [];
+    }
+
+    get reports() {
+        return this._reports;
+    }
+
+    get jsonReport() {
+        return {
+            id: this.id,
+            data: this.reportData || null,
+        };
+    }
+
+    get id() {
+        return this.constructor.name;
+    }
+
+    get active() {
+        if (!this._neededReports) return true;
+        for (const reportKey of this._neededReports) {
+            if (this.reports?.[reportKey] === undefined) return false;
+        }
+        return true;
+    }
+
+    render() {
+        if (this.props.mode === Story.MODES.DETAILS)
```

Comment:
Interesting - in `ReportStory` we have the two different render modes, but not in `Story`? I'd argue it belongs there, but if you wanted to refactor this in two steps, that works for me too.